editor.c

```diff
@@ -1,175 +1,298 @@
+	if(currNode->next!=NULL) 
+		currNode->next->prev = currNode->prev;
+	else
+		currNode->prev->next = NULL;
+	undoPush(currNode,'D',0); // when a new character is deleted it is pushed onto the undo stack
+	free(currNode);
+	return 1;
+}
+
+/*
+	Take the characters from the file and insert them into a string (fast loading).
+	Now construct the linkedlist from the string.
+	Return the head pointer of the Linked List.
+*/
+struct node* loadFileToList(char *filename)
+{
+	struct node* head = (struct node*) malloc(sizeof(struct node)); struct node* memory1;
+	head->prev = head->next = NULL;
+	char* list = malloc(100000); char ch;
+	int n = 0, i=0;
 	FILE *file;
-	char ch;
-	file=fopen(filename,"r");
-	
+	file = fopen(filename,"r+");
 	if (!file){
-		printf("File could not be opened. Press any key to continue! \n\a\a");
-		getchar();
-		return;
+		file = fopen(filename,"w+");
+		if(!file){
+			printf("File could not be opened. Press any key to continue! \n\a\a");
+			getchar();
+			return NULL;
+		}
 	}
-	
 	while((ch=fgetc(file))!=EOF){
-		append(&head, ch);
+		list[n++] = (char)ch;
 	}
-	
-	display(head);
-	return head;
 	fclose(file);
+	list[n] = '\0';
+	head->data = list[0];
+	while(list[i]!='\0'){
+		i++;
+		memory1 = (struct node*)malloc(sizeof(struct node));
+   		memory1->data = list[i];
+   		memory1->prev = head;
+   		memory1->next = NULL;
+   		head->next = memory1;
+	   	head = memory1;
+	}
+	printf("\nFile Loaded!\n");
+	return getHead(head);
 }
 
+//write the linked list back into the file
+void writeBackToFile(struct node* head_ref, char *filename)
+{
+	head_ref = getHead(head_ref);
+	FILE *file = fopen(filename,"w");
+	if (!file){		printf("File could not be opened. Press any key to continue! \n\a\a");		getchar();		return;	}
+	while(head_ref->next != NULL){
+		fputc(head_ref->data, file);
+		head_ref = head_ref->next;
+	}
+	fclose(file);
+}
+//API for cursor handling combining goRight() and goLeft()
+struct node* moveCursor(struct node* currNode, int n)
+{
+	struct node* temp = currNode;
+	if(n>0){
+		temp = goRight(temp, n);
+		undoPush(temp,'M',n); // when a move is made, it is pushed onto the undo stack
+	}
+	else{
+		temp = goLeft(temp, n);
+		undoPush(temp,'M',n); // when a move is made, it is pushed onto the undo stack
+	}
+	return temp;
+}
+/*
+	This function evaluates the undoRedoNode that has been popped off the top of the UR stack
+*/
+int evalURnode(struct undoRedoNode* undoRedoNode, struct node* currnode)
+{
+	// sees the operation in the undo node and takes action
+	/*
+		1. If the operation in the UR stack is insert then we delete the node at the current location.
+		2. If the operation in the UR stack is delete then we create a new node and insert the node
+		after the current location.
+		3. If the operation is move then we check the number of movements and its sign. Next we move
+		the cursor accordingly in the opposite direction.
+	*/
+	char operation1 = undoRedoNode->operation;
+	switch(operation1){
+		case 'I':
```

Comment:
We could use a full string for operations right? Like, `INSERT`, `DELETE`, `MOVE`, etc.
I suggest being more verbose here. What do you think?